crates/js-component-bindgen/src/transpile_bindgen.rs

```diff
@@ -546,6 +583,103 @@ impl<'a> Instantiator<'a, '_> {
         }
     }
     fn instantiate(&mut self) {
+        if self.use_asyncify {
+            let (if_async, if_async_await) =
+                if let Some(InstantiationMode::Sync) = self.gen.opts.instantiation {
+                    ("", "")
+                } else {
+                    ("async ", "await ")
+                };
+            uwrite!(self.src.js, "
```

Comment:
All of the functions and values written here should be moved into the `src/intrinsics.rs` file, rather than here on instantiation. It should be possible to write these functions to take all arguments explicitly instead of relying on closures? Or let me know if I'm missing something.

Not urgent of course, but noting now.

Reply:
Moved to the intrinsics